### PR TITLE
Support for $\backslash$ in file paths

### PR DESCRIPTION
**Pull request by koppor**
Fixes https://discourse.jabref.org/t/mendeley-bib-import-with-linked-files/3470. Fixes https://github.com/JabRef/jabref-issue-melting-pot/issues/178.

```[tasklist]
### Compulsory checks
- [ ] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [ ] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [ ] [Checked developer's documentation](https://devdocs.jabref.org/): Is the information available and up to date? If not, I outlined it in this pull request.
- [ ] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not, I created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, I submitted a pull request to the documentation repository.
```

### CODE REVIEW

File: src/main/java/org/jabref/logic/importer/util/FileFieldParser.java [hunk 1]
```diff
@@ -18,10 +18,15 @@ public class FileFieldParser {
...
     private StringBuilder charactersOfCurrentElement;
+
     private boolean windowsPath;
 
     public FileFieldParser(String value) {
-        this.value = value;
+        if (value == null) {
+            this.value = null;
+        } else {
+            this.value = value.replace("$\\backslash$", "\\");
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:28]: Why not call the latex 2 Unicode formatter to capture more cases

Comment (koppor) [reply src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:28]: I would need to rewrite the logic of the latextounicode formatter:

grafik

Expected :[ParsedFileField{description='', link='C:/Users/XXXXXX/AppData/Local/Mendeley Ltd./Mendeley Desktop/Downloaded/Brown - 2017 - Physical test methods for elastomers.pdf', fileType='pdf'}]
Actual   :[ParsedFileField{description='', link='Cbackslash{}{', fileType='}{/}Users/XXXXXX/AppData/Local/Mendeley Ltd./Mendeley Desktop/Downloaded/Brown - 2017 - Physical test methods for elastomers.pdf'}]

Expected :[ParsedFileField{description='test:;', link='wei2005ahp.pdf', fileType='PDF'}]
Actual   :[ParsedFileField{description='test  ', link='wei2005ahp.pdf', fileType='PDF'}]


Should I commit nevertheless?

Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:28]: I would go with the simple solution first and leave this whooe latex removing for later

Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:28]: One would probaly treat each path segment separately by the formatter

**Review by Siedlerchr — APPROVED** [review]

File: src/main/java/org/jabref/logic/importer/util/FileFieldParser.java [hunk 2]
```diff
@@ -18,10 +18,15 @@ public class FileFieldParser {
     private final String value;
 
     private StringBuilder charactersOfCurrentElement;
+
     private boolean windowsPath;
 
     public FileFieldParser(String value) {
-        this.value = value;
+        if (value == null) {
```
Comment (Siedlerchr) [inline src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:25]:
```suggestion
        if (value != null) {
         this.value = value.replace("$\\backslash$", "\\");

        }
        
```

Comment (koppor) [reply src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:25]: I don't see the diff properly. "value" is final. I wanted to keep it final. Is value assigned even if null with the proposal?

Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:25]: I didn't understand why you set the value to null again if it's already null

 if (value == null) {
value = null 
else
... 
was your code.

File: src/main/java/org/jabref/logic/importer/util/FileFieldParser.java (hunk 1 above)
Comment (koppor) [reply src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:28]: In the original example, there were no umlauts there. I was thinking: better a quick fix than some effort on catching the special cases 😇

File: src/main/java/org/jabref/logic/importer/util/FileFieldParser.java (hunk 2 above)
Comment (Siedlerchr) [reply src/main/java/org/jabref/logic/importer/util/FileFieldParser.java:25]: ah did not get your comment about the final. Okay.

**Review by Siedlerchr — APPROVED** [review]

**Pull request merged**